### PR TITLE
Feature/title component

### PR DESCRIPTION
**Pull request by rakeshkumar-verma**

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+	jcr:primaryType="cq:ClientLibraryFolder"
+    allowProxy="{Boolean}true"
+    categories="[hclecomm.title,hclecomm.content]"/>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml]: @rakeshkumar-verma : why doe we have 2 categories here ? can go with hclecomm.title

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+h6{
+&.cmp-title__text{
+font-size:16px;
+}
+}
+
+
+}
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less:48]: @rakeshkumar-verma : p tag will also have to be added here

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+less/title.less
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt:1]: @rakeshkumar-verma @ashishgoyalweb : what are the final variations for title ?

**Review by rishabh-behl — COMMENTED** [review]
@rakeshkumar-verma : pl check comments

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml (hunk 1 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml]: We can go but added hclecomm.content for all content component for future perspective. Let me know if want me remove hclecomm.content.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt (hunk 3 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt:1]: As per Acceptance criteria as of now will have H1 to H6 styles.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less (hunk 2 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less:48]: As of now in this component we have one H1-H6 element in html markup.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml (hunk 1 above)
Comment (rishabh-behl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml]: @rakeshkumar-verma : please remove . we can refer to same clientlib-  hclecomm.title,

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less (hunk 2 above)
Comment (rishabh-behl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less:48]: @rakeshkumar-verma : yes please add the p tag too in dropdown for title . it is required at various places

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt (hunk 3 above)
Comment (rishabh-behl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt:1]: @rakeshkumar-verma : will need the p tag too for this.

File: core/pom.xml [hunk 4]
```diff
@@ -188,5 +188,10 @@ Bundle-DocURL:
             <groupId>uk.org.lidalia</groupId>
             <artifactId>slf4j-test</artifactId>
         </dependency>
+		<!-- core component dependency added in core pom -->
+		<dependency>
+			<groupId>com.adobe.cq</groupId>
+			<artifactId>core.wcm.components.core</artifactId>
```
Comment (rishabh-behl) [inline core/pom.xml]: @rakeshkumar-verma : you would need to add the version of  core components too. Also why is this change part of title PR ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml (hunk 1 above)
Comment (rishabh-behl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml]: @rakeshkumar-verma : have you incorporated this change of removing hclecomm.content ?

File: core/src/main/java/com/hcl/ecomm/core/models/TitleModel.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
+package com.hcl.ecomm.core.models;
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/models/TitleModel.java]: @rakeshkumar-verma : why has the model been changed ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/_cq_dialog/.content.xml [hunk 6]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/_cq_dialog/.content.xml:1]: @rakeshkumar-verma : why have the chnages been made in the dialog . you could have modified teh design_dialog to add the p tag. refer screenshot below:

image

**Review by rishabh-behl — COMMENTED** [review]
@rakeshkumar-verma : pl incorporate comments

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml (hunk 1 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/.content.xml]: done

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less (hunk 2 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less:48]: added p tag

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt (hunk 3 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt:1]: default style added to p tag

File: core/pom.xml (hunk 4 above)
Comment (rakeshkumar-verma) [reply core/pom.xml]: removed.

File: core/src/main/java/com/hcl/ecomm/core/models/TitleModel.java (hunk 5 above)
Comment (rakeshkumar-verma) [reply core/src/main/java/com/hcl/ecomm/core/models/TitleModel.java]: removed

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/_cq_dialog/.content.xml (hunk 6 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/_cq_dialog/.content.xml:1]: review comment incorporated

Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/_cq_dialog/.content.xml:1]: review comment incorporated

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt (hunk 3 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/css.txt:1]: added p Tag

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less (hunk 2 above)
Comment (rakeshkumar-verma) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/title/clientlibs/less/title.less:48]: default style will apply on p tag

**Review by rishabh-behl — APPROVED** [review]
approved

**Comment by rishabh-behl** [comment]
merged

**Pull request merged**